### PR TITLE
pr_7

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

**Review by sns-seb2 — DISMISSED** [review]
SonarQube has detected 3 issues

File: src/main/java/org/sonarqube/A.java [hunk 1]
```diff
@@ -3,7 +3,11 @@
 public class A {
   String message = "foo";
 
+  // TODO fix this
   public String foo() {
+    message = null;
+    message.length();
+
     return message;
```
Comment (sns-seb2) [inline src/main/java/org/sonarqube/A.java:11]: ###AWAMQyi0S1b073eKUTZN###
org.sonarqube:pr-decoration:src/main/java/org/sonarqube/A.java:BRANCH:pr_7@9
A "NullPointerException" could be thrown; "message" is nullable here.

Comment (sns-seb2) [inline src/main/java/org/sonarqube/A.java:11]: ###AWAMQyizS1b073eKUTZM###
org.sonarqube:pr-decoration:src/main/java/org/sonarqube/A.java:BRANCH:pr_7@9
The return value of "length" must be used.

File: src/main/java/org/sonarqube/A.java [hunk 2]
```diff
@@ -3,7 +3,11 @@
 public class A {
   String message = "foo";
 
+  // TODO fix this
   public String foo() {
+    message = null;
```
Comment (sns-seb2) [inline src/main/java/org/sonarqube/A.java]: ###AWAMQyizS1b073eKUTZL###
org.sonarqube:pr-decoration:src/main/java/org/sonarqube/A.java:BRANCH:pr_7@6
Complete the task associated to this TODO comment.

**Comment by sns-seb** [comment]
analyzing again, no code change

**Comment by sns-seb** [comment]
no new review was posted, as expected
analyzing again, still no code change

**Review by sns-seb2 — DISMISSED** [review]
SonarQube has detected 2 issues

**Comment by sns-seb** [comment]
a new review was posted because the HEAD of the PR has changed, it includes no comment because all 2 issues have a non-outdated comment

**Comment by sns-seb** [comment]
analyzing again, no code change

**Comment by sns-seb** [comment]
as expected, no review posted

**Comment by sns-seb** [comment]
changing one issue to won't fix in SQ

**Comment by sns-seb** [comment]
status has been updated to "1 Bug", no change to comments nor review

**Comment by sns-seb** [comment]
analyzing again, no code change

**Comment by sns-seb** [comment]
as expected, no review posted (head of PR hasn't changed)

**Comment by sns-seb** [comment]
analyzing again, no code change, new review with even "Approved" expected...

**Comment by sns-seb** [comment]
as expected, no review posted

**Comment by sns-seb** [comment]
resolving the last bug as "won't fix"

**Review by sns-seb2 — DISMISSED** [review]

**Comment by sns-seb** [comment]
latest review was dismissed and a new one added with event "approved" \o/

**Comment by sns-seb** [comment]
reopening a bug...

**Comment by sns-seb** [comment]
status as been updated to "1 Bug" and no change to review nor comments (as expected)

**Comment by sns-seb** [comment]
analyzing PR again

**Review by sns-seb2 — DISMISSED** [review]
SonarQube has detected 1 issue

**Comment by sns-seb** [comment]
"Approved" review has been dismissed and new review added with even "changes_requested"

\o/

File: src/main/java/org/sonarqube/A.java (hunk 2 above)
Comment (sns-seb) [reply src/main/java/org/sonarqube/A.java]: simulating reply on outdated comment

File: src/main/java/org/sonarqube/A.java (hunk 1 above)
Comment (sns-seb) [reply src/main/java/org/sonarqube/A.java:11]: simulating reply on comment

Comment (sns-seb2) [reply src/main/java/org/sonarqube/A.java:11]: simulating auto reply on comment

File: src/main/java/org/sonarqube/A.java (hunk 2 above)
Comment (sns-seb2) [reply src/main/java/org/sonarqube/A.java]: simulating auto reply on outdated comment

**Review by sns-seb2 — CHANGES_REQUESTED** [review]
SonarQube has detected 1 issue

**Pull request closed without merging**